src/extras/MovieClip.js

```diff
@@ -191,40 +217,59 @@ MovieClip.prototype.gotoAndPlay = function (frameNumber)
  */
 MovieClip.prototype.update = function (deltaTime)
 {
+    var elapsed = this.animationSpeed * deltaTime;
 
-    this._currentTime += this.animationSpeed * deltaTime;
+    if (this._durations !== null)
+    {
+        var lag = this._currentTime % 1 * this._durations[this.currentFrame];
```

Comment:
`_currentTime` is a float. `1.24 % 1` returns `0.24`. It's a shortcut for `_currentTime - _currentTime | 0` I want to determine how long the current frame is already be shown.